Xamarin.Forms.Core/ListProxy.cs

```diff
@@ -213,14 +215,28 @@ void OnCollectionChanged(object sender, NotifyCollectionChangedEventArgs e)
 				sync.Callback(ProxiedEnumerable, sync.Context, () =>
 				{
 					e = e.WithCount(Count);
-					Device.BeginInvokeOnMainThread(action);
+					if (_listView != null)
+					{
+						Device.BeginInvokeOnMainThread(action, (_listView as BindableObject).WindowId);
+					}
+					else
+					{
+						Device.BeginInvokeOnMainThread(action);
+					}
```

Comment:
Rather than adding another member here (and making all the changes which go along with it), perhaps we can move the need to use a dispatcher out of `ListProxy` entirely. 

If we added a `CollectionChanged` event to `ListProxy` with an Action as one of the event arguments, we could listen for it in `ItemsView` and do the `BeginInvoke` there (since it's a `BindableObject` and thus has a `WindowId` associated with it). We might have to bubble the event up through another class (I think there's a `TemplatedItemsList` in between). But if we do that, we don't have to change all of the ListProxy test code.